apps/sms/views/conversation/js/conversation.js

```diff
@@ -2277,10 +2165,7 @@ var ConversationView = {
     this.enableConvertNoticeBanners();
 
     // If there was a draft, it just got sent so delete it.
-    if (this.draft) {
-      Drafts.delete(this.draft).store();
-      this.draft = null;
-    }
+    this.updateDraft();
```

Comment:
more importantly, I think cleanFields should really move away, see https://bugzilla.mozilla.org/show_bug.cgi?id=1136147#c4
We should call cleanFields only somewhat later (likely at "sending" event because we know the message was saved in gecko's db), and likely delete the draft at that moment as well.